### PR TITLE
Doc guidance on signing up for an issue by comment

### LINKED ISSUE #3029: Add guidance about how to sign up for solving an issue to the documentation
### Description

To reduce the likelihood of people working on the same issue at the same time there needs to be guidance provided to let people know how to `sign up`. The contribution docs should provide guidance on this.

### Additional context

None.

### PR DESCRIPTION
**Pull request by ehsteve**
### Description

Solves issue #3029 to provide guidance for developers to let people know that they are working on a particular issue.

Fixes #3029

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -75,20 +75,22 @@ Generally the more people that can look over a pull request the better it will t
 
 .. _pull requests: https://github.com/sunpy/sunpy/pulls
 
-Code
-----
+Code (to solve Issues)
```
Comment (nabobalis) [inline CONTRIBUTING.rst]: What if it's not an issue but the code is useful?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -75,20 +75,22 @@ Generally the more people that can look over a pull request the better it will t
...
+----------------------
 
 If you would prefer to code instead, we have several repositories you can investigate.
 The main one is the SunPy repository with where all the known `issues`_ with SunPy are detailed.
 Each issue should have a series of labels that provide information about nature of the issue.
+If you found an issue you'd like to work on, please make sure to add a comment on the issue
+to let people know that you are working on it!
```
Comment (nabobalis) [inline CONTRIBUTING.rst]: Any chance of removing the line break?

File: CONTRIBUTING.rst (hunk 1 above)
Comment (Cadair) [reply CONTRIBUTING.rst]: Yeah I am not sure we need this caveat here either.

Comment (ehsteve) [reply CONTRIBUTING.rst]: All the text below talks about working on issues which I thought was on purpose. If we want to expand the scope of this section, that's fine.

File: CONTRIBUTING.rst (hunk 2 above)
Comment (ehsteve) [reply CONTRIBUTING.rst]: That's a bridge too far!

File: CONTRIBUTING.rst (hunk 1 above)
Comment (nabobalis) [reply CONTRIBUTING.rst]: Yeah, I think adding a paragraph on what if you want to just contribute code upstream too would be good.

Comment (ehsteve) [reply CONTRIBUTING.rst]: Done.

File: CONTRIBUTING.rst (hunk 2 above)
Comment (ehsteve) [reply CONTRIBUTING.rst]: Done.

**Comment by nabobalis** [comment]
You will hate me but please can you only break each sentence on a new line and not midsentance?

**Comment by ehsteve** [comment]
@nabobalis I will only do that if you can point to me where that is requirement is described in the contribution guide for docs and if it does not exist then show me your PR to fix that :P

**Comment by nabobalis** [comment]
Yes that is a good point, that line should be in the dev guide's doc file and not the contributing guide. 

We do have an open issue https://github.com/sunpy/sunpy/issues/2849 that no one pushed back on.

**Comment by nabobalis** [comment]
it be easier if I just fix it in a separate PR and not worry about it here.

**Comment by nabobalis** [comment]
Thanks @ehsteve

**Pull request merged**